### PR TITLE
fix: More migration fixes, bumping contracts

### PR DESCRIPTION
**Pull request by kelemeno**
## Checklist

- [ ] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.
- [ ] Code has been formatted via `zk fmt` and `zk lint`.
- [ ] Spellcheck has been run via `zk spellcheck`.
- [ ] Linkcheck has been run via `zk linkcheck`.

### CODE REVIEW

File: core/tests/ts-integration/package.json [hunk 1]
```diff
@@ -1,36 +1,36 @@
...
-        "typescript": "^4.3.5",
-        "zksync-ethers": "https://github.com/zksync-sdk/zksync-ethers#sb-old-sdk",
-        "elliptic": "^6.5.5"
-    }
+  "name": "ts-integration",
+  "version": "0.1.0",
+  "license": "MIT",
+  "private": true,
+  "scripts": {
+    "test": "zk f jest --forceExit --testTimeout 120000",
```
Comment (StanislavBreadless) [inline core/tests/ts-integration/package.json:7]: why return to 120k?

Comment (StanislavBreadless) [inline core/tests/ts-integration/package.json:7]: also, why did we change the padding for this file?

File: docs/specs/zk_evm/vm_specification/compiler/instructions/evm/environment.md [hunk 2]
```diff
@@ -12,7 +12,7 @@ is common for Yul and EVMLA representations.
 ### System Contract
 
 This information is requested a System Contract called
-[L2BaseToken](https://github.com/code-423n4/2023-10-zksync/blob/main/code/system-contracts/contracts/L2BaseToken.sol).
+[L2BaseToken](https://github.com/code-423n4/2023-10-zksync/blob/main/code/system-contracts/contracts/L2EthToken.sol).
```
Comment (StanislavBreadless) [inline docs/specs/zk_evm/vm_specification/compiler/instructions/evm/environment.md:15]: doesnt look correct

File: infrastructure/protocol-upgrade/src/hyperchain-upgrade.ts [hunk 3]
```diff
@@ -58,6 +61,21 @@ async function hyperchainUpgrade1() {
     );
 }
 
+async function insertAddresses() {
```
Comment (StanislavBreadless) [inline infrastructure/protocol-upgrade/src/hyperchain-upgrade.ts:64]: why do we need it? we will have the new facets deployed and they would be written to the facets.json

File: infrastructure/protocol-upgrade/src/hyperchain-upgrade.ts [hunk 4]
```diff
@@ -66,7 +84,22 @@ async function hyperchainUpgrade2() {
...
+    const cwd = process.cwd();
+    process.chdir(`${process.env.ZKSYNC_HOME}/contracts/l1-contracts/`);
+
+    await spawn(`yarn hyperchain-upgrade-3 | tee deployL1.log`);
+    process.chdir(cwd);
+}
+
 async function preparePostUpgradeCalldata() {
+    console.log(
+        'Preparing post upgrade calldata',
```
Comment (StanislavBreadless) [inline infrastructure/protocol-upgrade/src/hyperchain-upgrade.ts:97]: I would suggest to either remove the log or make sure that each outputted value has corresponding text before it so it is easier to understand without looking into the code

File: infrastructure/protocol-upgrade/src/hyperchain-upgrade.ts [hunk 5]
```diff
@@ -81,20 +114,121 @@ async function preparePostUpgradeCalldata() {
     fs.writeFileSync(postUpgradeCalldataFileName, JSON.stringify(calldata, null, 2));
 }
 
+async function deploySharedBridgeL2Implementation() {
+    const cwd = process.cwd();
+    process.chdir(`${process.env.ZKSYNC_HOME}/contracts/l2-contracts/`);
+
+    await spawn(`yarn deploy-shared-bridge-l2-implementation  | tee deployL1.log`);
```
Comment (StanislavBreadless) [inline infrastructure/protocol-upgrade/src/hyperchain-upgrade.ts:121]: nit 
```suggestion
    await spawn(`yarn deploy-shared-bridge-l2-implementation | tee deployL1.log`);
```

Comment (StanislavBreadless) [inline infrastructure/protocol-upgrade/src/hyperchain-upgrade.ts:121]: probably a different file should be chosen to write the output to

File: infrastructure/protocol-upgrade/src/hyperchain-upgrade.ts [hunk 6]
```diff
@@ -81,20 +114,121 @@ async function preparePostUpgradeCalldata() {
...
+    );
+    await spawn('zk f yarn  workspace protocol-upgrade-tool start crypto save-verification-params');
+    await preparePostUpgradeCalldata();
+    await spawn(
+        `zk f yarn  workspace protocol-upgrade-tool start transactions build-default --upgrade-timestamp 1711451944 --zksync-address ${process.env.CONTRACTS_DIAMOND_PROXY_ADDR} --use-new-governance --upgrade-address ${process.env.CONTRACTS_HYPERCHAIN_UPGRADE_ADDR} --post-upgrade-calldata`
+    );
+    await spawn(
+        `zk f yarn  workspace protocol-upgrade-tool start transactions propose-upgrade --zksync-address ${process.env.CONTRACTS_DIAMOND_PROXY_ADDR} --new-governance ${process.env.CONTRACTS_GOVERNANCE_ADDR}`
+    );
+    // await spawn(
```
Comment (StanislavBreadless) [inline infrastructure/protocol-upgrade/src/hyperchain-upgrade.ts:169]: commented out code

File: infrastructure/protocol-upgrade/src/hyperchain-upgrade.ts [hunk 7]
```diff
@@ -81,20 +114,121 @@ async function preparePostUpgradeCalldata() {
...
+    .option('--shared-bridge-l2-implementation')
     .option('--phase1')
+    .option('--insert-addresses')
     .option('--post-upgrade-calldata')
     .option('--phase2')
+    .option('--phase3')
+    .option('--full-start')
+    .option('--post-propose')
+    .option('--execute-upgrade')
+    // .option('--get-diamond-upgrade-batch-number-for-eth-withdrawals')
```
Comment (StanislavBreadless) [inline infrastructure/protocol-upgrade/src/hyperchain-upgrade.ts:196]: comented out code

File: infrastructure/protocol-upgrade/src/hyperchain-upgrade.ts [hunk 8]
```diff
@@ -81,20 +114,121 @@ async function preparePostUpgradeCalldata() {
...
         } else if (options.phase2) {
             await hyperchainUpgrade2();
         } else if (options.postUpgradeCalldata) {
             await preparePostUpgradeCalldata();
+        } else if (options.sharedBridgeL2Implementation) {
+            await deploySharedBridgeL2Implementation();
+        } else if (options.phase3) {
+            await hyperchainUpgrade3();
+        } else if (options.getDiamondUpgradeBatchNumberForEthWithdrawals) {
+            // What we care about for withdrawals is the executed batch number, since we are storing isWithdrawalFinalized flag, for some valid executed txs.
```
Comment (StanislavBreadless) [inline infrastructure/protocol-upgrade/src/hyperchain-upgrade.ts:213]: no printing logic is there yet

File: infrastructure/protocol-upgrade/src/hyperchain-upgrade.ts [hunk 9]
```diff
@@ -81,20 +114,121 @@ async function preparePostUpgradeCalldata() {
...
+            // this is the batch number that the last deposit is processed in ( this is tied to a tx, so commit vs executed is not relevant)
+            // we will print the priority tx queue id as part of phase 2 script for local testing, and we can use that to find the batch number
+            // for the mainnet upgrade we will have to manually check the priority queue at the given block, since governance is signing the txs
+        } else if (options.fullStart) {
+            await hyperchainFullUpgrade();
+        } else if (options.postPropose) {
+            await postPropose();
+        } else if (options.executeUpgrade) {
+            await spawn(
+                `zk f yarn  workspace protocol-upgrade-tool start transactions execute-upgrade --zksync-address ${process.env.CONTRACTS_DIAMOND_PROXY_ADDR} --new-governance ${process.env.CONTRACTS_GOVERNANCE_ADDR}`
```
Comment (StanislavBreadless) [inline infrastructure/protocol-upgrade/src/hyperchain-upgrade.ts:231]: nit

```suggestion
                `zk f yarn workspace protocol-upgrade-tool start transactions execute-upgrade --zksync-address ${process.env.CONTRACTS_DIAMOND_PROXY_ADDR} --new-governance ${process.env.CONTRACTS_GOVERNANCE_ADDR}`
```

File: infrastructure/zk/src/contract.ts [hunk 10]
```diff
@@ -129,12 +129,19 @@ export async function deployL2ThroughL1({ includePaymaster }: { includePaymaster
...
     const l2DeploymentEnvVars = [
         'CONTRACTS_L2_SHARED_BRIDGE_ADDR',
+        'CONTRACTS_L2_ERC20_BRIDGE_ADDR',
         'CONTRACTS_L2_TESTNET_PAYMASTER_ADDR',
         'CONTRACTS_L2_WETH_TOKEN_IMPL_ADDR',
         'CONTRACTS_L2_WETH_TOKEN_PROXY_ADDR',
         'CONTRACTS_L2_DEFAULT_UPGRADE_ADDR'
     ];
     updateContractsEnv(`etc/env/l2-inits/${process.env.ZKSYNC_ENV!}.init.env`, l2DeployLog, l2DeploymentEnvVars);
+    // erc20 bridge is now deployed as shared bridge, but we still need the config var:
```
Comment (StanislavBreadless) [inline infrastructure/zk/src/contract.ts:139]: why do we need it? it is optional in the server

File: yarn.lock
Comment (StanislavBreadless) [inline yarn.lock:1]: Why are these changes needed?

File: core/tests/ts-integration/package.json (hunk 1 above)
Comment (kelemeno) [reply core/tests/ts-integration/package.json:7]: I just changed the padding because it also changed in kl-factory, so just changing it back.

 the 120k is an error, will fix

Comment (kelemeno) [reply core/tests/ts-integration/package.json:7]: https://github.com/matter-labs/zksync-era/pull/298/files#r1572483926

File: docs/specs/zk_evm/vm_specification/compiler/instructions/evm/environment.md (hunk 2 above)
Comment (kelemeno) [reply docs/specs/zk_evm/vm_specification/compiler/instructions/evm/environment.md:15]: This was needed for link check, as the codeArena is the old code arena. Alternatively the new one could be used.

File: infrastructure/protocol-upgrade/src/hyperchain-upgrade.ts (hunk 3 above)
Comment (kelemeno) [reply infrastructure/protocol-upgrade/src/hyperchain-upgrade.ts:64]: I am using a different deployment script, because we also need to deploy the STM, Bridgehub, and other contracts, and we need to set the initial diamond cut in the STM. Doing this seemed easier.

File: infrastructure/zk/src/contract.ts (hunk 10 above)
Comment (kelemeno) [reply infrastructure/zk/src/contract.ts:139]: We need it to test the old SDK,

File: yarn.lock
Comment (kelemeno) [reply yarn.lock:1]: should I not update yarn.lock when I run yarn? Its not that they are needed, I thought committing this was the correct policy.

File: docs/specs/zk_evm/vm_specification/compiler/instructions/evm/environment.md (hunk 2 above)
Comment (kelemeno) [reply docs/specs/zk_evm/vm_specification/compiler/instructions/evm/environment.md:15]: I can put this if you think it is better: 
https://github.com/code-423n4/2024-03-zksync/blob/main/code/system-contracts/contracts/L2BaseToken.sol

Comment (StanislavBreadless) [reply docs/specs/zk_evm/vm_specification/compiler/instructions/evm/environment.md:15]: I would either:
- Copy from 2024-03 branch
- Or refrain from editing docs at all in this PR

File: yarn.lock
Comment (StanislavBreadless) [reply yarn.lock:1]: @kelemeno it was an issue in the kl-factory branch that `yarn.lock` is updated after each `yarn` run

Comment (StanislavBreadless) [reply yarn.lock:1]: it will was pointed out by Danil and will be resolved here https://github.com/matter-labs/zksync-era/pull/1745

File: infrastructure/protocol-upgrade/src/hyperchain-upgrade.ts (hunk 8 above)
Comment (kelemeno) [reply infrastructure/protocol-upgrade/src/hyperchain-upgrade.ts]: I added the printing logic

**Review by StanislavBreadless — APPROVED** [review]
